compiler/rustc_middle/src/mir/consts.rs

```diff
@@ -230,6 +230,20 @@ impl<'tcx> Const<'tcx> {
         ))
     }
 
+    #[inline]
+    pub fn is_required_const(&self) -> bool {
+        // Only unevalated consts must be added to `required_consts` as only those can possibly
+        // still have latent const-eval errors.
+        match self {
+            Const::Ty(c) => match c.kind() {
+                ty::ConstKind::Param(_) | ty::ConstKind::Error(_) | ty::ConstKind::Value(_) => true,
```

Comment:
This used to be `false`. But I don't understand why. If it's `ConstKind::Error`, then it clearly error'd, so we better make sure this body is not evaluated, don't we? (Otherwise it ICEs in `tests/ui/transmutability/malformed-program-gracefulness/wrong-type-assume.rs`.)

And if it's `Param`, then it could get substituted with something that errors later, no?